### PR TITLE
144-FEATURE-Improve-code-formatting-and-linting

### LINKED ISSUE #144: [FEATURE] Improve code formatting and linting
### Issue to be solved

There is currently no unified approach to code formatting and linting. This makes it hard to collaborate on the code base. Saving a file on one machine may completely change the formatting of the whole file. This makes it difficult to see which parts of the code properly were altered when reviewing PRs. Although the project has a prettier and eslint setting, there is no editor config which forces the format on save option. That's produced code that is not uniformly formatted - some lines are formatted one way, and other lines in the same file are formatted another

### Suggest A Solution

1. Tighten up the prettier and eslint settings. They are currently too loose
2.  Introduce an [editor config](https://editorconfig.org/) file that forces format on save
3. Add a GitHub action that prevents code from being checked in or merged if it does not follow the formatting rules

### PR DESCRIPTION
**Pull request by tombeckenham**
I've created a setup for linting and prettier. Note that I have run prettier on all files, but not fixed any lint errors. That would be best to do on a file by file basis

Fixes #144

### CODE REVIEW

**Comment by tombeckenham** [comment]
NOTE: Do not merge this PR yet. First check if these settings work with your editor and general coding style. If so, we need to run prettier on every single file in the repo before merging - it will change almost every file which will make it hard to figure out changes

**Review by tombeckenham — COMMENTED** [review]
This can now be merged when ready

**Review by tombeckenham — COMMENTED** [review]
When approved I mean!

**Review by zzggo — APPROVED** [review]

**Pull request merged**
